pkg/kube/krt/index.go

```diff
@@ -42,18 +44,20 @@ func (i IndexObject[K, O]) ResourceName() string {
 
 // NewNamespaceIndex is a small helper to index a collection by namespace
 func NewNamespaceIndex[O Namespacer](c Collection[O]) Index[string, O] {
-	return NewIndex(c, func(o O) []string {
+	return NewIndex(c, cache.NamespaceIndex, func(o O) []string {
 		return []string{o.GetNamespace()}
 	})
 }
 
-// NewIndex creates a simple index, keyed by key K, over an informer for O. This is similar to
+// NewIndex creates a simple index, keyed by key K, over a collection for O. This is similar to
 // Informer.AddIndex, but is easier to use and can be added after an informer has already started.
+// Different collection implementations may reuse existing indexes with the same name.
```

Comment:
Might be worth noting: two different informer collections _do_ share the same index. This differs from other collections

Reply:
added clarification for this, let me know if it's enough